### PR TITLE
Support some /deployments/:id/whitelist commands

### PR DESCRIPTION
**Pull request by camelpunch**
# What

* POST /2016-07/deployments/:id/whitelist
* GET /2016-07/deployments/:id/whitelist

# Why

Needed for https://github.com/alphagov/paas-cf/pull/1151

# Who can't

@camelpunch nor @henrytk nor @chrisfarms

### CODE REVIEW

File: whitelist.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+// Create a single deployment whitelist entry for a CIDR range
+func (c *Client) CreateDeploymentWhitelist(deploymentID string, params DeploymentWhitelistParams) (*Recipe, []error) {
+	body, errs := c.createDeploymentWhitelistJSON(deploymentID, params)
+	if errs != nil {
+		return nil, errs
+	}
+
+	deployed := Recipe{}
+	json.Unmarshal([]byte(body), &deployed)
```
Comment (bandesz) [inline whitelist.go]: there is no error handling

File: whitelist.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+	return &deployed, nil
+}
+
+func (c *Client) getWhitelistForDeploymentJSON(deploymentid string) (string, []error) {
```
Comment (bandesz) [inline whitelist.go:90]: correct variable name: ```deploymentID```

File: whitelist.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	return c.getJSON("deployments/" + deploymentid + "/whitelist")
+}
+
+// Get all whitelist entries for a given deployment ID
+func (c *Client) GetWhitelistForDeployment(deploymentid string) ([]DeploymentWhitelist, []error) {
```
Comment (bandesz) [inline whitelist.go]: correct variable name: ```deploymentID```

File: whitelist.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	body, errs := c.getWhitelistForDeploymentJSON(deploymentid)
+	if errs != nil {
+		return nil, errs
+	}
+
+	whitelistResponse := deploymentWhitelistResponse{}
+	json.Unmarshal([]byte(body), &whitelistResponse)
```
Comment (bandesz) [inline whitelist.go]: there is no error handling here

**Review by bandesz — CHANGES_REQUESTED** [review]

File: whitelist.go (hunk 3 above)
Comment (camelpunch) [reply whitelist.go]: This was chosen for consistency with e.g. [deployment.go](https://github.com/alphagov/paas-gocomposeapi/blob/29792fdf68bea79c8c46ef6eb16df897c9b161f4/deployment.go#L186), but agree that we can just have grown-up names here.

File: whitelist.go (hunk 4 above)
Comment (camelpunch) [reply whitelist.go]: Again, these were copy+pasta'd from the other bits of the code. I think it'd be better to sort this out with tests at this level. We'd then want another PR to back-fill error handling for creating deployments etc. too.

Comment (bandesz) [reply whitelist.go]: I would still sort out error handling in this PR for this use case, and tests would be even better, but I didn't request them originally as there are no tests in the library atm.

**Comment by camelpunch** [comment]
We've chosen to keep consistency of variable naming with the rest of the codebase. Perhaps we'll PR in a change to make these variables adhere to the Go naming conventions in the future, across the board in the project.

**Review by bandesz — APPROVED** [review]

**Pull request merged**